### PR TITLE
EPMDJ-7309: Implement useStorageHook

### PR DESCRIPTION
**Pull request by RomanAchilov**

### CODE REVIEW

File: src/common/util/local-storage.ts [hunk 1]
```diff
@@ -8,3 +9,19 @@ export async function save(data: any) {
   const storage = await browser.storage.local.get();
   await browser.storage.local.set({ ...storage, ...data });
 }
+
+type Keys = string | string[] | Record<string, any> | null;
+
+export const getStorage = async (keys: Keys): Promise<Record<string, any>> => new Promise((resolve, reject) => {
```
Comment (RomanDavlyatshin) [inline src/common/util/local-storage.ts:15]: @RomanAchilov
Use the `src/common/util/promisify-browser-api-call.ts` utility wrapper.
```
export const getStorage = promisifyBrowserApiCall(chrome.storage.local.get)
```
See exemplary usage at `src/background/dev-tools-api.ts`

File: src/hooks/use-chrome-storage.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+export const useChromeStorage = <T>(key: string, initialValue: T) => {
+  const [storedValue, setStoredValue] = useState<T>(initialValue);
+
+  useEffect(() => {
+    const handleStateChange = (changes: { [key: string]: chrome.storage.StorageChange }) => setStoredValue(changes[key].newValue);
+
+    chrome.storage.onChanged.addListener(handleStateChange);
+    const storageValue = getStorage([key]);
+    storageValue.then((result) => {
```
Comment (RomanDavlyatshin) [inline src/hooks/use-chrome-storage.ts:12]: Refactor to `async/await` syntax (in line with the rest of the codebase 😸 ).
You can utilize async IIFE to get around `useEffect` restriction on passing the async functions.

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]
Looks good, but tbh, I haven't tested it yet 😄 
I have some minor issues with the boilerplate code/style

File: src/common/util/local-storage.ts (hunk 1 above)
Comment (RomanAchilov) [reply src/common/util/local-storage.ts:16]: With this solution I don't need to bind the context as in the previous commit

File: src/common/util/local-storage.ts [hunk 3]
```diff
@@ -10,18 +11,16 @@ export async function save(data: any) {
...
+      return reject(chrome.runtime.lastError);
+    }
+    resolve(items);
+  });
 });
 
-export const getStorage = promisifyChromeApiCall<Record<string, any>, Keys | Callback>(chrome.storage.local.get.bind(chrome.storage.local));
-export const addToStorage = promisifyChromeApiCall<any, any>(chrome.storage.local.set.bind(chrome.storage.local));
+// does it make sense to promisify this method?
+export const addToStorage = (data: Record<string, any>) => chrome.storage.local.set(data);
```
Comment (RomanAchilov) [inline src/common/util/local-storage.ts:26]: does it make sense to promisify this method?

**Pull request closed without merging**